app/common/prescription/controllers/patientSimplifiedPrescriptionController.js

```diff
@@ -370,45 +398,97 @@
         prescription.prescriptionItems.push(prescriptionItem);
       });
 
-      if(validateBeforeSave(prescription)){
-        prescriptionService.create(prescription)
+      if(validateCreatePrescription(form, prescription)){
+
+           prescriptionService.create(prescription)
           .then(function () {
             notifier.success($filter('translate')('COMMON_MESSAGE_SUCCESS_ACTION_COMPLETED'));
             vm.listedPrescriptions = [];
+            vm.selectedProvider =  { display: '' };
             isPrescriptionControl();
+            getProviderAndPrescriptionDateModal().modal('hide');
             spinner.forPromise(loadSavedPrescriptions(patient));
           })
           .catch(function () {
             notifier.error($filter('translate')('COMMON_MESSAGE_COULD_NOT_CREATE_PRESCRIPTION'));
           });
       }
-    }
-
-    function validateBeforeSave(prescription){
+      else{
+        vm.selectedProvider = { display: '' };
+      }
+     }
 
-      var isArvPrescriptionToBeCreated = false;
-      _.forEach(prescription.prescriptionItems, function (prescriptionItem) {
-           if(prescriptionItem.regime){
-             isArvPrescriptionToBeCreated = true;
-           }
-      });
+    function validateCreatePrescription(form, prescription){
 
-        var hasExistingArvPrescription = false;
-       _.forEach(vm.existingPrescriptions, function (prescription) {
-         _.forEach(prescription.prescriptionItems, function (prescriptionItem) {
-            if( prescription.prescriptionStatus == 'ACTIVE' && prescriptionItem.regime){
-            hasExistingArvPrescription = true;
-            }
-         });
-        });
+      if(form.selectedProvider.$invalid)
+      {
+        return false;
+      }
 
-       if(isArvPrescriptionToBeCreated && hasExistingArvPrescription){
-         notifier.error($filter('translate')('COMMON_MESSAGE_COULD_NOT_CREATE_ARV_PRESCRIPTION_BECAUSE_EXISTS_AN_ACTIVE_ARV_PRESCRIPTION'));
+      if(hasActiveArvPrescriptionForNewArvItem(prescription)){
+        notifier.error($filter('translate')('COMMON_MESSAGE_COULD_NOT_CREATE_ARV_PRESCRIPTION_BECAUSE_EXISTS_AN_ACTIVE_ARV_PRESCRIPTION'));
+        return false;
+      }
+      var duplicatedItems = [];
+      duplicatedItems = hasDuplicatedExistingActivePrescriptionItems(prescription, duplicatedItems);
```

Comment:
Looks like it is not necessary to pass  ```duplicatedItems``` as a function parameter. I thing its better to use it like this ```duplicatedItems = hasDuplicatedExistingActivePrescriptionItems(prescription);```